### PR TITLE
Be more robust when determining if is charm or bundle. 

### PR DESCRIPTION
**Pull request by bac**
Also fix tests for new version of charm.v6-unstable.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore-client/153/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore-client/154/

File: cmd/charm/charmcmd/push.go [hunk 1]
```diff
@@ -141,16 +141,18 @@ func (c *pushCommand) Run(ctxt *cmd.Context) error {
...
-		if err == nil {
-			break
-		}
-		if b1, err1 := charm.ReadBundleDir(srcDir); err1 == nil {
-			b = b1
-			err = nil
-			c.id.Series = "bundle"
+		if charm.IsCharmDir(srcDir) {
+			ch, err = charm.ReadCharmDir(srcDir)
+			if err == nil {
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/push.go]: This if statement doesn't seem to do anything.

Comment (bac) [reply cmd/charm/charmcmd/push.go]: You're right. Left over from previous flow.

**Review by fabricematrat — APPROVED** [review]

File: cmd/charm/charmcmd/push.go (hunk 1 above)
Comment (rogpeppe) [reply cmd/charm/charmcmd/push.go]: Did you intend to do:

     if charm.IsCharmDir(srcDir) {
        ch, err = charm.ReadCharmDir(srvDir)
        break
    }
    if b, err = charm.ReadBundleDir(srvDir); err == nil {
        c.id.Series = "bundle"
    }


?

Although I actually don't see why we're setting c.id.Series here at all - why not just do:

     if charm.IsCharmDir(srcDir) {
        ch, err = charm.ReadCharmDir(srvDir)
     } else {
         b, err = charm.ReadBundleDir(srvDir)
     }

and set c.id.Series inside the b != nil case below?

**Review by rogpeppe — APPROVED** [review]
LGTM modulo one suggestion.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore-client/155/

**Comment by bac** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-client-merge

**Pull request merged**